Packs/FeedUnit42v2/Integrations/FeedUnit42v2/FeedUnit42v2.py

```diff
@@ -144,27 +164,125 @@ def parse_indicators(indicator_objects: list, feed_tags: Optional[list] = None,
+                        obj_refs_excluding_relationships_prefix.append(related_obj)
+                        if id_to_object.get(related_obj):
+                            entity_b_obj_type, entity_b_value = STIX2XSOARParser.get_entity_b_type_and_value(related_obj,
+                                                                                                             id_to_object,
+                                                                                                             True)
+                            relationship = EntityRelationship(
+                                name='related-to',
+                                entity_a=f"[Unit42 ATOM] {report_object.get('name')}",
+                                entity_a_type=ThreatIntel.ObjectsNames.REPORT,
+                                entity_b=entity_b_value,
+                                entity_b_type=entity_b_obj_type
+                            ).to_indicator()
+                            relationships.append(relationship)
+    if obj_refs_excluding_relationships_prefix:
+        object_ref_to_return.extend([{'objectstixid': object} for object in obj_refs_excluding_relationships_prefix])
+    return relationships, object_ref_to_return
+
+
+def is_atom42_main_report(report_obj):
+    """ Our definition for main report is a report with 'object_refs'
+    list that contains only objects of the type not a report and a report with
+    description.
 
+    Args:
+      report_obj: The report object.
+    Returns:
+        A boolean.
+    """
+    obj_refs = report_obj.get('object_refs', [])
+    contain_report = False
+    contain_intrusion_set = False
+    if report_obj.get('name') == 'ATOM Campaign Report':
+        return False
+    for obj_ref in obj_refs:
+        if not obj_ref.startswith(('report--', 'intrusion-set--')):
+            return False
+        if obj_ref.startswith('intrusion-set--'):
+            contain_intrusion_set = True
+        if obj_ref.startswith('report--'):
+            contain_report = True
+    return contain_intrusion_set and contain_report
+
+
+def is_atom42_sub_report(report_obj):
+    """ Our definition for sub report is a report with 'object_refs'
+    list that contains only objects not for type intrusion-set or report.
 
-def is_sub_report(report_obj):
+    Args:
+      report_obj: The report object.
+    Returns:
+        A boolean.
+    """
     obj_refs = report_obj.get('object_refs', [])
-    return all(not obj_ref.startswith('report--') for obj_ref in obj_refs)
+    is_report_description = report_obj.get('description') is not None
+    only_reports_and_intrusion_set = True
+    if report_obj.get('name') == 'ATOM Campaign Report':
+        return True
+    for obj_ref in obj_refs:
+        if not obj_ref.startswith(('report--', 'intrusion-set--')):
+            only_reports_and_intrusion_set &= False
+    return (not is_report_description) and (not only_reports_and_intrusion_set)
```

Comment:
```suggestion
    return is_report_description or only_reports_and_intrusion_set
```
I think it works (de morgan rules)

Reply:
Correct!